### PR TITLE
[App Config] `az appconfig`: Add warning log info to output when the `--yes` flag is set 

### PR DESCRIPTION
**Pull request by albertofori**
**Related command**

`az appconfig import`
`az appconfig export`

**Description**

Related to [Azure/AppConfiguration issue #712](https://github.com/Azure/AppConfiguration/issues/712). This fix separates kv diffing from print previews and enables logging of specific information to output even when the `--yes` flag is set.

**Testing Guide**

**History Notes**

[App Config] `az appconfig import/export`: Add warning log info to output even when `--yes` flag is set

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [x] The PR title and description has followed the guideline in [Submitting Pull Requests](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests).

- [x] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

- [x] I adhere to the [Error Handling Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/error_handling_guidelines.md).

### CODE REVIEW

**Comment by yonzhan** [comment]
App Config

**Comment by albertofori** [comment]
Hi @avanigupta, Could you kindly take a look at this? Thanks!

File: src/azure-cli/azure/cli/command_modules/appconfig/_kv_helpers.py [hunk 1]
```diff
@@ -1079,31 +1124,30 @@ def __import_kvset_from_file(client, path, strict, yes):
...
-        if not changes_detected:
-            logger.warning('Target configuration store already contains all configuration settings in source. No changes will be made.')
-            return
+    existing_kvset_list = __serialize_kv_list_to_comparable_json_list(existing_kvset, ImportExportProfiles.KVSET)
+    kvset_to_import_list = __serialize_kv_list_to_comparable_json_list(kvset_to_import, ImportExportProfiles.KVSET)
 
+    changes_detected = __print_kvset_json_diff(existing_kvset_list, kvset_to_import_list, yes=yes)
+    
+    if not changes_detected:
+        logger.warning('Target configuration store already contains all configuration settings in source. No changes will be made.')
```
Comment (avanigupta) [inline src/azure-cli/azure/cli/command_modules/appconfig/_kv_helpers.py:1147]: Would this laready be logged by __print_kvset_json_diff?

File: src/azure-cli/azure/cli/command_modules/appconfig/_constants.py [hunk 2]
```diff
@@ -72,3 +72,8 @@ class KVSetConstants:
 class ImportExportProfiles:
     KVSET = "appconfig/kvset"
     DEFAULT = "appconfig/default"
+
+class ImportExportDiffKeys:
```
Comment (avanigupta) [inline src/azure-cli/azure/cli/command_modules/appconfig/_constants.py:76]: We are using 3 forms of the verbs here: 
- `update` is returned by differ output
- `updated`  is used as an intermediate key
- `Updating` is what the user sees.

We can't change the first one, and the last one makes more sense to users. So can we use either of these forms for the intermediate dict key?
My preference is change it to `update' and the class can be called JsonDiffResult.
So the usage would look like this:
```
if action.label == JsonDiffResult.UPDATE:
            ff_diff[JsonDiffResult.UPDATE] = []
```

Comment (avanigupta) [reply src/azure-cli/azure/cli/command_modules/appconfig/_constants.py:76]: Or just `JsonDiff.Update`

File: src/azure-cli/azure/cli/command_modules/appconfig/_kv_helpers.py (hunk 1 above)
Comment (albertofori) [reply src/azure-cli/azure/cli/command_modules/appconfig/_kv_helpers.py]: My bad! This has been updated. Thanks!

File: src/azure-cli/azure/cli/command_modules/appconfig/_constants.py (hunk 2 above)
Comment (albertofori) [reply src/azure-cli/azure/cli/command_modules/appconfig/_constants.py]: Sounds good! This simplifies it. This has been updated. Thanks!

**Review by avanigupta — DISMISSED** [review]

File: src/azure-cli/azure/cli/command_modules/appconfig/_kv_helpers.py [hunk 3]
```diff
@@ -689,28 +739,23 @@ def __print_preview(old_json, new_json, strict=False):
     res = differ.diff(old_json, new_json)
     keys = str(res.keys())
     if res == {} or (('update' not in keys) and ('insert' not in keys) and (not strict or ('delete' not in keys))):
-        logger.warning('\nTarget configuration already contains all key-values in source. No changes will be made.')
-        return False
+        return kv_diff
```
Comment (yanzhudd) [inline src/azure-cli/azure/cli/command_modules/appconfig/_kv_helpers.py:742]: Could you please add a comment for this if statement?

**Comment by zhoxing-ms** [comment]
Please note that we will launch the release for this sprint tomorrow morning. Since there are still some comments above, so we postpone the release time of this PR to the next sprint **(04-04)**, could you accept this release time?

**Comment by albertofori** [comment]
> > [App Config] BREAKING CHANGE: az appconfig import: Log info to output even when --yes flag is set
> > [App Config] BREAKING CHANGE: az appconfig export: Log info to output even when --yes flag is set
> 
> I'd like to confirm with you why it is marked as a breaking change? Will printing the log cause a breaking change to the user's script?

@zhoxing-ms  Yes please. For users who have scripts that should terminate on warnings, this could potentially cause the script to break. Just a precaution on this.

**Comment by albertofori** [comment]
> Please note that we will launch the release for this sprint tomorrow morning. Since there are still some comments above, so we postpone the release time of this PR to the next sprint **(04-04)**, could you accept this release time?

Since the only outstanding ones are for adding comments, could we make the release tomorrow if I address those comments and push the changes now?

**Comment by albertofori** [comment]
Hi @zhoxing-ms. I just wanted to confirm if this will make the release window as the comments have been addressed.

**Comment by zhoxing-ms** [comment]
> Yes please. For users who have scripts that should terminate on warnings, this could potentially cause the script to break. Just a precaution on this.

OK, so I'd like to confirm with you that the breaking change contained in this PR only refers to the warning log at present?
If so, in fact, the warning log is not strictly a breaking change for CLI, because it will not directly affect the input, output and command's behavior

**Comment by Anino1996** [comment]
> > Yes please. For users who have scripts that should terminate on warnings, this could potentially cause the script to break. Just a precaution on this.
> 
> OK, so I'd like to confirm with you that the breaking change contained in this PR only refers to the warning log at present? If so, in fact, the warning log is not strictly a breaking change for CLI, because it will not directly affect the input, output and command's behavior

Yes please. That’s the only change here. The input and returned output behaviour does not change for users.

**Comment by zhoxing-ms** [comment]
> Hi @zhoxing-ms. I just wanted to confirm if this will make the release window as the comments have been addressed.

I'm so sorry for my late reply. I didn't see your comments in time because I was super busy last Friday.
Because this PR was created late *(we usually recommend to submit PR one week before the code freeze date, which can leave us enough time to review and improve PR)* and may contain breaking changes, we did not release it in this sprint for the sake of the impact on users.
If the breaking change contained in this PR only refers to the warning log, we don't need to wait for the next breaking change release. We will release it at the next sprint **(04-04)** as soon as possible, may I ask could you accept this release time?

**Comment by albertofori** [comment]
> > Hi @zhoxing-ms. I just wanted to confirm if this will make the release window as the comments have been addressed.
> 
> I'm so sorry for my late reply. I didn't see your comments in time because I was super busy last Friday. Because this PR was created late _(we usually recommend to submit PR one week before the code freeze date, which can leave us enough time to review and improve PR)_ and may contain breaking changes, we did not release it in this sprint for the sake of the impact on users and [breaking change process](https://eng.ms/docs/cloud-ai-platform/azure-core/azure-management-and-platforms/control-plane-bburns/azure-cli-tools-azure-cli-powershell-and-terraform/azure-cli-tools/teams_docs/azcli_docs/breaking_changes). Since the breaking change contained in this PR only refers to the warning log, we don't need to wait for the next breaking change release. We will release it at the next sprint **(04-04)** as soon as possible, may I ask could you accept this release time?

@zhoxing-ms No worries. The 04-04 release time should be fine. Thanks!

**Comment by zhoxing-ms** [comment]
> No worries. The 04-04 release time should be fine. Thanks!

Thanks for your understanding~

**Comment by zhoxing-ms** [comment]
Since we usually do not add the breaking change tag for `logger.warning`, So I modified your PR title and history notes

**Review by zhoxing-ms — APPROVED** [review]

**Review by yanzhudd — APPROVED** [review]

**Pull request merged**